packages/core/src/shared/lsp/lspResolver.ts

```diff
@@ -31,32 +33,38 @@ export class LanguageServerResolver {
      * @throws ToolkitError if no compatible version can be found
      */
     async resolve() {
-        const latestVersion = this.latestCompatibleLspVersion()
-        const targetContents = this.getLSPTargetContents(latestVersion)
-        const cacheDirectory = this.getDownloadDirectory(latestVersion.serverVersion)
-
-        const serverResolvers: StageResolver<LspResult>[] = [
-            {
-                resolve: async () => await this.getLocalServer(cacheDirectory, latestVersion, targetContents),
-                telemetryMetadata: { id: this.lsName, languageServerLocation: 'cache' },
-            },
-            {
-                resolve: async () => await this.fetchRemoteServer(cacheDirectory, latestVersion, targetContents),
-                telemetryMetadata: { id: this.lsName, languageServerLocation: 'remote' },
-            },
-            {
-                resolve: async () => await this.getFallbackServer(latestVersion),
-                telemetryMetadata: { id: this.lsName, languageServerLocation: 'fallback' },
-            },
-        ]
-
-        return await tryStageResolvers('getServer', serverResolvers, getServerVersion)
-
+        const timeout = new Timeout(5000)
+        await showMessageWithCancel(`Downloading '${this.lsName}' language server`, timeout)
```

Comment:
Is the cancel handled? I.e. it should actually cancel the download. Look for existing `showMessageWithCancel` to see examples.

When the user hits Cancel, it should actually cancel the download.